lightning/src/ln/functional_test_utils.rs

```diff
@@ -3374,6 +3387,33 @@ pub fn create_node_cfgs_with_persisters<'a>(node_count: usize, chanmon_cfgs: &'a
 	nodes
 }
 
+pub fn create_node_cfgs<'a>(node_count: usize, chanmon_cfgs: &'a Vec<TestChanMonCfg>) -> Vec<NodeCfg<'a>> {
+	let persisters = chanmon_cfgs.iter().map(|c| &c.persister).collect();
+	create_node_cfgs_internal(node_count, chanmon_cfgs, persisters, |graph, keys_manager| {
+		test_utils::TestMessageRouter::new_default(graph, keys_manager)
+	})
+}
+
+pub fn create_node_cfgs_with_persisters<'a>(
+	node_count: usize,
+	chanmon_cfgs: &'a Vec<TestChanMonCfg>,
+	persisters: Vec<&'a impl test_utils::SyncPersist>,
+) -> Vec<NodeCfg<'a>> {
+	create_node_cfgs_internal(node_count, chanmon_cfgs, persisters, |graph, keys_manager| {
+		test_utils::TestMessageRouter::new_default(graph, keys_manager)
+	})
+}
+
+pub fn create_node_cfgs_with_node_id_message_router<'a>(
+	node_count: usize,
+	chanmon_cfgs: &'a Vec<TestChanMonCfg>,
+) -> Vec<NodeCfg<'a>> {
+	let persisters = chanmon_cfgs.iter().map(|c| &c.persister).collect();
+	create_node_cfgs_internal(node_count, chanmon_cfgs, persisters, |graph, keys_manager| {
+		test_utils::TestMessageRouter::new_node_id_router(graph, keys_manager)
+	})
+}
```

Comment:
nit: For these, you don't need to pass a closure because the constructors take the same arguments as the expected closure type. So just pass `test_utils::TestMessageRouter::new_default`, for example.

Reply:
Ah! Thanks for the pointer!
Updated in [pr3246.23](https://github.com/shaavan/rust-lightning/commits/pr3246.23)